packages/fxa-content-server/app/scripts/views/pair/auth_totp.js

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import AuthErrors from 'lib/auth-errors';
+import Cocktail from 'cocktail';
+import FormView from '../form';
+import ServiceMixin from '../mixins/service-mixin';
+import Template from '../../templates/sign_in_totp_code.mustache';
+import VerificationReasonMixin from '../mixins/verification-reason-mixin';
+import FlowEventsMixin from '../mixins/flow-events-mixin';
+
+const CODE_INPUT_SELECTOR = 'input.totp-code';
+
+const View = FormView.extend({
+  className: 'pair-totp',
+  template: Template,
+
+  getAccount () {
+    return this.model.get('account') || this.getSignedInAccount();
+  },
+
+  submit () {
+    const account = this.getAccount();
+    const code = this.getElementValue('input.totp-code');
+    return account.verifyTotpCode(code, 'pair')
+      .then((result) => {
+        if (result.success) {
+          return this.replaceCurrentPage('/pair/auth/allow', {totpComplete: true});
```

Comment:
@rfk thoughts on this? 

proposal is:

* we always do TOTP for pairing if account has 2fa (verifyTotpCode)
* when `totpComplete` transitions, one check to add is when the `allow` page is `if TOTP is enabled on the account AND /totp/exists reports true` then let it through. (This is not in this PR just yet)

Reply:
Seems reasonable to me